modules/post/windows/gather/enum_unattend.rb

```diff
@@ -99,8 +95,7 @@ def save_raw(xmlpath, data)
   def get_registry_unattend_path
     # HKLM\System\Setup!UnattendFile
     begin
-      key = session.sys.registry.open_key(HKEY_LOCAL_MACHINE, 'SYSTEM')
-      fname = key.query_value('Setup!UnattendFile').data
+      fname = registry_getvaldata("HKEY_LOCAL_MACHINE\\System\\Setup", "UnattendFile").strip
```

Comment:
This might be wrong because my system didn't had the `UnattendFile` registry key and I am not sure if it always saves like `HKEY_LOCAL_MACHINE\System\Setup /v "UnattendFile"`.

Reply:
This looks correct. `registry_getval_data` uses `query_value_direct` in the Meterpreter implementation.

Reply:
okay :+1:

Reply:
Actually, I'm getting a crash here when this registry key does not exist because the return value is `nil`.

```
[-] Post failed: NoMethodError undefined method `strip' for nil:NilClass
[-] Call stack:
[-]   /home/smcintyre/Repositories/metasploit-framework.pr/modules/post/windows/gather/enum_unattend.rb:98:in `get_registry_unattend_path'
[-]   /home/smcintyre/Repositories/metasploit-framework.pr/modules/post/windows/gather/enum_unattend.rb:135:in `init_paths'
[-]   /home/smcintyre/Repositories/metasploit-framework.pr/modules/post/windows/gather/enum_unattend.rb:142:in `run'
[*] Post module execution completed
```

I tested that with both a shell and Meterpreter session. In both cases it crashes.